core/src/main/scala/org/apache/spark/sql/sources/StoreStrategy.scala

```diff
@@ -52,7 +54,16 @@ object StoreStrategy extends Strategy {
     case drop: DropTable =>
       ExecutedCommandExec(drop) :: Nil
 
-    case DMLExternalTable(name, storeRelation: LogicalRelation, insertCommand) =>
+    case p: EncoderPlan[_] =>
+      val plan = p.asInstanceOf[EncoderPlan[Any]]
+      EncoderScanExec(plan.rdd.asInstanceOf[RDD[Any]],
+        plan.encoder, plan.isFlat, plan.output) :: Nil
+
+    case logical.InsertIntoTable(l@LogicalRelation(p: PlanInsertableRelation,
```

Comment:
Wouldn't StoreStrategy change the CreateTableUsingAsSelect into ExecutedCommandExec which will not use the insert route?